test/specs/map/MapService.spec.js

```diff
@@ -1271,127 +1271,242 @@ describe('ga_map_service', function() {
+        layer.bodId = 'toto';
+        expect(gaLayerFilters.potentialTooltip(layer)).to.be(false);
+      });
     });
 
-    it('potentialTooltip keeps visible layers in the display manager',
-        function() {
-      var layer = new ol.layer.Tile();
-      gaDefinePropertiesForLayer(layer);
-      layer.displayInLayerManager = true;
-      expect(!!gaLayerFilters.potentialTooltip(layer)).to.be(false);
-      layer.visible = true;
-      expect(!!gaLayerFilters.potentialTooltip(layer)).to.be(false);
-      layer.bodId = 'toto';
-      expect(!!gaLayerFilters.potentialTooltip(layer)).to.be(true);
+    describe('searchable', function() {
+      var layer, stub;
 
-      var layer = new ol.layer.Tile();
-      gaDefinePropertiesForLayer(layer);
-      layer.displayInLayerManager = true;
-      expect(!!gaLayerFilters.potentialTooltip(layer)).to.be(false);
-      layer.visible = true;
-      expect(!!gaLayerFilters.potentialTooltip(layer)).to.be(false);
-      layer.bodId = 'samere';
-      expect(!!gaLayerFilters.potentialTooltip(layer)).to.be(false);
-
-      // Not a vector layer -> no support of intersection on vector layers
-      // in query tool
-      layer = new ol.layer.Vector({ source: new ol.source.Vector({}) });
-      gaDefinePropertiesForLayer(layer);
-      layer.displayInLayerManager = true;
-      expect(!!gaLayerFilters.potentialTooltip(layer)).to.be(false);
-      layer.visible = true;
-      expect(!!gaLayerFilters.potentialTooltip(layer)).to.be(false);
-      layer.bodId = 'toto';
-      expect(!!gaLayerFilters.potentialTooltip(layer)).to.be(false);
+      beforeEach(function() {
+        stub = sinon.stub(gaLayers, 'getLayerProperty');
+        layer = new ol.layer.Tile();
+        gaDefinePropertiesForLayer(layer);
+        layer.displayInLayerManager = true;
+      });
+
+      afterEach(function() {
+        stub.restore();
+      });
+
+      it('keeps searchable and visible bod layers', function() {
+        stub.returns(true);
+        expect(gaLayerFilters.searchable(layer)).to.be(false);
+        layer.visible = true;
+        expect(gaLayerFilters.searchable(layer)).to.be(false);
+        layer.bodId = 'toto';
+        expect(gaLayerFilters.searchable(layer)).to.be(true);
+      });
+
+      it('excludes bod layer not searchable', function() {
+        stub.returns(false);
+        layer.visible = true;
+        layer.bodId = 'toto';
+        expect(gaLayerFilters.searchable(layer)).to.be(false);
+      });
+    });
+
+    describe('queryable', function() {
+      var layer, stub;
+
+      beforeEach(function() {
+        stub = sinon.stub(gaLayers, 'getLayerProperty');
+        layer = new ol.layer.Tile();
+        gaDefinePropertiesForLayer(layer);
+        layer.displayInLayerManager = true;
+      });
+
+      afterEach(function() {
+        stub.restore();
+      });
+
+      it('keeps queryable and visible bod layers', function() {
+        stub.returns(['attr1']);
```

Comment:
For my own understanding this means that `gaLayers.getLayerProperty` will always return ['attr1'] ?
`